### PR TITLE
Allow IPv6LinkLocalAddressGenerationMode in networkConfig

### LINKED ISSUE #199407: IPv6LinkLocalAddressGenerationMode option is not supported in networkConfig section
```
        networkConfig = {
           IPv6LinkLocalAddressGenerationMode = "random";
         };
```

```
trace: Systemd Network has extra fields [IPv6LinkLocalAddressGenerationMode].
trace: Systemd Network has extra fields [IPv6LinkLocalAddressGenerationMode].
error: A definition for option `systemd.network.networks."81-name".networkConfig' is not of type `attribute set of (systemd option)'.
```

This option is documented [here](https://www.freedesktop.org/software/systemd/man/systemd.network.html#IPv6LinkLocalAddressGenerationMode=)

### PR DESCRIPTION
**Pull request by jpathy**
Allow option `IPv6LinkLocalAddressGenerationMode` in `systemd.network.networks.<name>.networkConfig`.

Fixes: #199407 .

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bb010g — APPROVED** [review]

**Review by 9ary — APPROVED** [review]

**Comment by bb010g** [comment]
Bump @dasJ.

**Comment by dasJ** [comment]
@ofborg eval

**Comment by 9ary** [comment]
Bump again. However, considering [RFC 0042](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md), I do wonder if keeping all of the asserts really is all that valuable. I think the maintenance burden mostly outweighs the benefits.

**Comment by spacekitteh** [comment]
Could this be merged please @dasJ

**Review by Majiir — APPROVED** [review]
Patch works as advertised.

@jpathy the commit comment does not fit `CONTRIBUTING.md`, which might be why nobody has merged this yet. You could update it to something like:

```
nixos/networkd: allow IPv6LinkLocalAddressGenerationMode in networkConfig
```

**Comment by bb010g** [comment]
> the commit comment does not fit `CONTRIBUTING.md`, which might be why nobody has merged this yet. You could update it to something like:
> 
> ```
> nixos/networkd: allow IPv6LinkLocalAddressGenerationMode in networkConfig
> ```

If a commit message is all that's holding this up, a maintainer can easily rewrite that along with the merge. That shouldn't be the reason this has been languishing for almost a whole year with no other problems.

**Review by mkg20001 — APPROVED** [review]

**Pull request merged**
